File: app/src/main/java/ch/bailu/aat/views/preferences/AbsSelectOverlayDialog.java
package ch.bailu.aat.views.preferences;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

import ch.bailu.aat.preferences.SolidOverlayFileList;
import ch.bailu.util_java.foc.Foc;


public abstract class AbsSelectOverlayDialog extends AbsSolidDialog implements  DialogInterface.OnClickListener {

    private final SolidOverlayFileList slist;

    public AbsSelectOverlayDialog(Context context) {
        this(new SolidOverlayFileList(context));
    }

    public AbsSelectOverlayDialog (SolidOverlayFileList l) {
        slist=l;

        final AlertDialog.Builder dialog = createDefaultDialog(l);
        dialog.setItems(slist.getStringArray(), this);
        dialog.create();
        dialog.show();
    }




    @Override
    public void onClick(DialogInterface dialog, int i) {
        Foc file = slist.get(i).getValueAsFile();
        onFileSelected(file);
        dialog.dismiss();
    }

    protected abstract void onFileSelected(Foc file);
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/dialog/AbsSelectOverlayDialog.kt
package ch.bailu.aat.views.preferences.dialog

import android.content.Context
import android.content.DialogInterface
import ch.bailu.aat_lib.app.AppContext
import ch.bailu.aat_lib.preferences.map.overlay.SolidCustomOverlay
import ch.bailu.aat_lib.preferences.map.overlay.SolidOverlayList
import ch.bailu.foc.Foc

abstract class AbsSelectOverlayDialog(appContext: AppContext, context: Context, private val slist: SolidOverlayList<SolidCustomOverlay> = SolidOverlayList.createCustomOverlayList(appContext)
): AbsSolidDialog(), DialogInterface.OnClickListener {
    init {

        val dialog = createDefaultDialog(context, slist)
        dialog.setItems(slist.getStringArray(), this)
        dialog.create()
        dialog.show()
    }

    override fun onClick(dialog: DialogInterface, index: Int) {
        val file = slist[index].getValueAsFile()
        onFileSelected(slist, index, file)
        dialog.dismiss()
    }

    protected abstract fun onFileSelected(slist: SolidOverlayList<SolidCustomOverlay>, index: Int, file: Foc)
}
